src/hotspot/cpu/riscv/c2_MacroAssembler_riscv.cpp

```diff
@@ -2163,27 +2163,74 @@ void C2_MacroAssembler::enc_cmove(int cmpFlag, Register op1, Register op2, Regis
 
 // Set dst to NaN if any NaN input.
 void C2_MacroAssembler::minmax_fp(FloatRegister dst, FloatRegister src1, FloatRegister src2,
-                                  bool is_double, bool is_min) {
+                                  FLOAT_TYPE ft, bool is_min) {
+  assert_cond((ft != FLOAT_TYPE::half_precision) || UseZfh);
+
   Label Done, Compare;
 
-  is_double ? fclass_d(t0, src1)
-            : fclass_s(t0, src1);
-  is_double ? fclass_d(t1, src2)
-            : fclass_s(t1, src2);
+  switch (ft) {
+    case FLOAT_TYPE::half_precision:
```

Comment:
I wouldn't switch three times on **ft**.

Maybe just switch once, and have all in that switch, e.g.:
```
    case FLOAT_TYPE::half_precision:
      fclass_h(t0, src1);
      fclass_h(t1, src2);
      orr(t0, t0, t1);
      andi(t0, t0, FClassBits::nan); // if src1 or src2 is quiet or signaling NaN then return NaN
      fadd_h(dst, src1, src2);
      beqz(t0, Compare);
      j(Done);
      bind(Compare);
      if (is_min) {
        fmin_h(dst, src1, src2);
      } else {
        fmax_h(dst, src1, src2);
      }
      break;
```

No ?

Reply:
Good suggestion, seems it's better.